packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -429,7 +429,15 @@ async function startWatcher(opts: SetupOpts) {
           pagesType: isAppPath ? PAGE_TYPES.APP : PAGE_TYPES.PAGES,
         })
 
-        if (isAppPath && isMetadataRoute(pageName)) {
+        if (
+          isAppPath &&
```

Comment:
nit: `isAppPath` already checks `appDir` so only the second one is needed I think.

Reply:
`appDir` here is only to make ts happy, `isAppPath` cannot ensure `appDir` is truthy.

Reply:
but is `isAppPath` being used by anything inside of this `if`?

Reply:
thought was the opposite, will follow up later in other PRs